components/net/fetch/methods.rs

```diff
@@ -317,7 +320,32 @@ fn basic_fetch(request: Rc<Request>, cache: &mut CORSCache) -> Response {
             }
         },
 
-        "blob" | "file" | "ftp" => {
+        "file" => {
+            if *request.method.borrow() == Method::Get {
+                match url.to_file_path() {
+                    Ok(file_path) => {
+                        let file_path_dup = &file_path.to_owned();
+                        if let Some(mut file) = File::open(file_path).ok() {
```

Comment:
We could also potentially use `.map_or` here, so the whole thing would look something like:
```rust
File::open(file_path.clone()).ok().map_or(Response::network_error(), |mut file| {
    let mut bytes = vec![];
    let _ = file.read_to_end(&mut bytes);
    let mime = guess_mime_type(file_path);

    let mut response = Response::new();
    *response.body.lock().unwrap() = ResponseBody::Done(bytes);
    response.headers.set(ContentType(mime));
    response
})